### PR TITLE
Add support for multiple VoiceSets on an actor [Depends on sound setup rewrite]

### PR DESCRIPTION
**Pull request by penev92**
As wished by @DoGyAUT and company.
If multiple VoiceSets are provided, one will be chosen at random when the actor is created, and will be used for the entire lifetime of that actor.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Voiced.cs [hunk 1]
```diff
@@ -32,22 +30,23 @@ public class Voiced : IVoiced
 	{
 		public readonly VoicedInfo Info;
 
+		public string VoiceSet { get; }
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/Voiced.cs]: I'm pretty sure `{ get; }` is a getter-only auto-property which was introduced in C# 6 (which OpenRA doesn't support yet).

Comment (atlimit8) [inline OpenRA.Mods.Common/Traits/Voiced.cs]: Why not `public readonly string VoiceSet;`?

**Comment by pchote** [comment]
It would be nice if we could use upgrades to turn certain sets on and/or off.

**Comment by GraionDilach** [comment]
This could work nicely with factions as well - solving the RA2 Engineer voice problem.

**Comment by RoosterDragon** [comment]
Needs a rebase.

File: OpenRA.Mods.Common/Traits/Voiced.cs (hunk 1 above)
Comment (penev92) [inline OpenRA.Mods.Common/Traits/Voiced.cs]: Because it implements it from the interface, and I don't want to mess with that more than I need to here.

**Comment by penev92** [comment]
Rebased. The upgrade ideas do have potential, but are out of scope here, and also need to have the methodology cleared out (check if enabled in `PlaySound()`, or when picking a voice set, or what).

**Comment by abcdefg30** [comment]
Needs a rebase again.

File: OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs [hunk 2]
```diff
@@ -2833,7 +2833,7 @@ internal static void UpgradeActorRules(int engineVersion, ref List<MiniYamlNode>
 				if (engineVersion < 20160107 && depth == 1 && node.Key.StartsWith("Cloak"))
 				{
 					var defaultCloakType = Traits.UncloakType.Attack
-						| Traits.UncloakType.Unload | Traits.UncloakType.Infiltrate | Traits.UncloakType.Demolish;
+					                       | Traits.UncloakType.Unload | Traits.UncloakType.Infiltrate | Traits.UncloakType.Demolish;
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs]: This change doesn't look correct to me, please revert it.

**Comment by penev92** [comment]
Rebased.

**Comment by penev92** [comment]
As was pointed out on IRC today ([link](http://logs.openra.net/?year=2016&month=02&day=06#15:55:09)), this can be (kinda) achieved with the voice set's `Variants`.
It is a hacky and unpleasant approach IMO, but will have to do for now until we rewrite the audio setup
>[18:03] <+pchote> the alternative would be to remove the legacy implementation
[18:03] <+pchote> and transition the current mods to the new setup
[18:04] <+pchote> penev: it would make your pr the first part of the sound rewrite
...
[18:09] <+penev> anyway, if we're going to transition from the current glued-together-with-duck-tape-and-gum audio setup to sound sequences
...
[18:18] <+penev> ok so I guess that would want to be closed until the audio re-setup

**Pull request closed without merging**